pkg/config/flags.go

```diff
@@ -37,5 +37,18 @@ func newFlagSet() *flag.FlagSet {
 	flagSet.String("auth-admin-role", "", "The name of the only role that will be passed on the request if it is found in the list of roles")
 	flagSet.String("auth-default-role", "", "The role given to every request unless it has the auth-admin-role")
 
+	//net/http.Server timeouts for the server side of the proxy
+	flagSet.Duration("http-read-timeout", time.Duration(1)*time.Minute, "The maximum duration for reading the entire HTTP request. Zero means no timeout.")
```

Comment:
Should we expand the flags to identify the unit? e.g `http-read-timeout-sec`

Reply:
That was my first impulse, too, but I don't think it's the right thing. A flag of type time.Duration accepts many different time units, from nanoseconds to hours. See https://golang.org/pkg/flag/#Duration. Under the covers, https://golang.org/pkg/time/#ParseDuration does the parsing.

Reply:
Can i define a proper duration in configuration on the CLI?  It will let me use say `1s` or `10m`?  If so, I'm good as-is

Reply:
It will even let you define `--http-expect-continue-timeout=1h2m3s4ms5us6ns` if you so desire.